optimum/intel/openvino/quantization.py

```diff
@@ -676,7 +676,12 @@ def _prepare_builtin_dataset(self, quantization_config: OVWeightQuantizationConf
             quantization_config.tokenizer, trust_remote_code=quantization_config.trust_remote_code
         )
         nsamples = quantization_config.num_samples if quantization_config.num_samples else 128
-        calibration_dataset = get_dataset(quantization_config.dataset, tokenizer, seqlen=32, nsamples=nsamples)
+        if isinstance(quantization_config.dataset, str):
+            calibration_dataset = get_dataset(quantization_config.dataset, tokenizer, seqlen=32, nsamples=nsamples)
+        else:
```

Comment:
@nikita-savelyevv, can you please invest more time here. First, we need to check that this is a List of strings. Second, I don't like the name of the function where this logic is located: `_prepare_builtin_dataset`

Reply:
> First, we need to check that this is a List of strings.

Added the check

> Second, I don't like the name of the function where this logic is located: _prepare_builtin_dataset

Renamed to `_prepare_causal_lm_dataset`. What do you think?

Reply:
It could be `_prepare_text_dataset` but I am also ok with your name.